packages/core/src/components/renvooi/renvooi.tsx

```diff
@@ -1,31 +1,72 @@
-import { Component, ComponentInterface, Fragment, FunctionalComponent, Prop, h } from "@stencil/core";
+import {
+  Component,
+  ComponentInterface,
+  Fragment,
+  FunctionalComponent,
+  Prop,
+  h,
+  Event,
+  EventEmitter,
+} from "@stencil/core";
 
-import { RenvooiValue } from "./renvooi.interfaces";
+import {
+  RenvooiMarkFunction,
+  RenvooiMarkItemHighlightEvent,
+  RenvooiRenderMarkFunction,
+  RenvooiValue,
+} from "./renvooi.interfaces";
 
 interface RenvooiRenderProps {
   value: RenvooiValue;
+  mark?: RenvooiRenderMarkFunction;
+  emitMarkItemHighlight(text: string, elementRef: HTMLElement): void;
 }
 
-const RenvooiRender: FunctionalComponent<RenvooiRenderProps> = ({ value }) => {
+function renderText(text: string, emit: (text: string, ref: HTMLElement) => void, mark?: RenvooiRenderMarkFunction) {
+  if (!mark) {
+    return text;
+  }
+
+  const result = mark(text);
+
+  if (!result) {
+    return text;
+  }
+
+  return result.map((renvooiText) => {
+    if (typeof renvooiText === "string") {
+      return <>{renvooiText}</>;
+    }
+
+    return (
+      <mark
+        class={renvooiText.highlight ? "dso-highlight" : undefined}
+        ref={(ref) => renvooiText.highlight && ref && emit(renvooiText.text, ref)}
+      >
+        {renvooiText.text}
+      </mark>
+    );
+  });
+}
+
+const RenvooiRender: FunctionalComponent<RenvooiRenderProps> = ({ value, mark, emitMarkItemHighlight }) => {
   if (typeof value === "string" || !value) {
     // This element is used for --_dso-renvooi-text-decoration
-    return <span class="text">{value}</span>;
+    return <span class="text">{renderText(value, emitMarkItemHighlight, mark)}</span>;
```

Comment:
Heb je overwogen om `renderText()` een functional component te maken?

```suggestion
    return <span class="text"><RenderText value={value} onMarkItemHighlight={} /></span>;
```

Een functie die JSX teruggeeft, en in een `.tsx` file wordt gebruikt klinkt als iets wat een functional component moet zijn

Reply:
Niet overwogen. Maar heb het nu wel gedaan.

Reply:
Leuk. Ben benieuwd